sign_language_datasets/datasets/dgs_corpus/dgs_corpus.py

```diff
@@ -208,7 +209,19 @@ def _generate_examples(self, data):
             if self._builder_config.include_video:
                 videos = {t: str(datum["video_" + t]) if ("video_" + t) in datum else "" for t in ["a", "b", "c"]}
 
-                features["fps"] = self._builder_config.fps if self._builder_config.fps is not None else default_fps
+                fps_expected_by_loader = self._builder_config.fps if self._builder_config.fps is not None else default_fps
+
+                for video_path in videos:
+                    if video_path == "":
+                        continue
+
+                    # make sure that the video fps is as expected
+                    actual_video_fps = cv2.VideoCapture(video_path).get(cv2.CAP_PROP_FPS)
+
+                    assert actual_video_fps == float(fps_expected_by_loader), \
```

Comment:
The loader allows to load custom fps, for example, `fps=10`. This is passed via `ffmpeg` and is loaded [here](https://github.com/sign-language-processing/datasets/blob/master/sign_language_datasets/datasets/config.py#L42) indeed in 10 fps.

I think the check should be to compare the videos always to `default_fps`.

Reply:
I don't understand this yet. If `self._builder_config.fps` is defined, shouldn't it take precedence?